src/hooks/usePermitAllowance.test.ts

```diff
@@ -0,0 +1,129 @@
+import { Contract } from '@ethersproject/contracts'
+import { MaxAllowanceTransferAmount, PERMIT2_ADDRESS } from '@uniswap/permit2-sdk'
+import { CurrencyAmount } from '@uniswap/sdk-core'
+import { UNIVERSAL_ROUTER_ADDRESS } from '@uniswap/universal-router-sdk'
+import PERMIT2_ABI from 'abis/permit2.json'
+import { Permit2 } from 'abis/types'
+import { SupportedChainId } from 'constants/chains'
+import { UNI } from 'constants/tokens'
+import { useSingleCallResult } from 'hooks/multicall'
+import { useContract } from 'hooks/useContract'
+import ms from 'ms.macro'
+import { renderHook, waitFor } from 'test'
+
+import { usePermitAllowance, useUpdatePermitAllowance } from './usePermitAllowance'
+
+const OWNER = hardhat.account.address
+const SPENDER = UNIVERSAL_ROUTER_ADDRESS(SupportedChainId.MAINNET)
+
+const TOKEN = UNI[SupportedChainId.MAINNET]
+const EXPIRATION = 1234567890
+const NONCE = 42
+
+jest.mock('hooks/multicall')
+jest.mock('hooks/useContract')
+const mockUseContract = useContract as jest.Mock
+const mockUseSingleCallResult = useSingleCallResult as jest.Mock
+
+describe('usePermitAllowance', () => {
+  const CONTRACT = new Contract(PERMIT2_ADDRESS, PERMIT2_ABI) as Permit2
```

Comment:
nit: can these live top level?

Reply:
done. also moved some other (related) const's to the top level.